gnmidiff/cmd/root.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cmd contains a CLI utility to access gnmidiff functionality.
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func Execute() {
+	rootCmd := &cobra.Command{
+		Use:   "gnmidiff",
+		Short: "gnmidiff is a utility for comparing between SetRequests and Notifications",
+	}
+
+	cfgFile := rootCmd.PersistentFlags().String("config_file", "", "Path to config file.")
+	rootCmd.PersistentPreRunE = func(cmd *cobra.Command, args []string) error {
+		if *cfgFile != "" {
+			viper.SetConfigFile(*cfgFile)
+			if err := viper.ReadInConfig(); err != nil {
+				return fmt.Errorf("error reading config: %w", err)
+			}
+		}
+		viper.BindPFlags(cmd.Flags())
+		viper.AutomaticEnv()
+		return nil
+	}
+
+	rootCmd.AddCommand(newSetRequestDiffCmd())
+
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
```

Comment:
i think its cleaner if the main() calls os.Exit(), i prefer either
1. the func returns to the root command and you call cmd.Execute in the main
2. return an error here and set this exit code in the main

Reply:
Done.